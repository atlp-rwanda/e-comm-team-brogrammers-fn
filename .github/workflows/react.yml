name: React.js CI/CD

on:
  push:
    branches: ['develop']
  pull_request:
    branches: ['develop']
    types: [opened, synchronize, reopened]

env:
  REACT_APP_SERVER_URL: https://brogrammers4.onrender.com

jobs:
  build-and-deploy-preview:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [16.x, 18.x]
        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/

    steps:
      - uses: actions/checkout@v3
      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v3
        with:
          node-version: ${{ matrix.node-version }}
          cache: 'npm'
      - run: npm i
      - run: npm test
      - run: npm run build
      - name: Deploy Preview
        uses: actions/setup-node@v3
        with:
          environment: preview
          ref: ${{ github.event.pull_request.head.sha }}
          deploy_key: ${{ secrets.DEPLOY_KEY }}
        if: github.event.pull_request.reviews == null
