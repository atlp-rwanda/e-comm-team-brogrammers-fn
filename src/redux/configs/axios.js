import ax from 'axios';

const axios = ax.create({
  baseURL: process.env.REACT_APP_SERVER_URL,
});
export default axios;
