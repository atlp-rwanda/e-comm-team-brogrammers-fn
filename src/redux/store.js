/* eslint-disable import/no-named-as-default */
import { configureStore } from '@reduxjs/toolkit';
import booleanReducer from './features/slices/sample';
import loginReducer from './features/slices/login';
import userReducer from './features/slices/user';
import signupReducer from './features/slices/signup';
import logoutReducer from './features/slices/logout';
import passwordReducer from './features/slices/password';
import productSlice from './features/slices/products';
import categoriesReducer from './features/slices/categories';
import addItemReducer from './features/slices/additem';
import searchReducer from './features/slices/searchslice';
import collectionSlice from './features/slices/sellerCollection';
import oneProductSlice from './features/slices/oneProduct';
import notificationSlice from './features/slices/notifications';
import reviewSlice from './features/slices/productReview';
import viewWishlistSlice from './features/slices/wishlistslice';
import clearWishlistSlice from './features/slices/clearwishlist';
import addToWishlistSlice from './features/slices/addWishlistslice';
import DeleteToWishlistSlice from './features/slices/deleteWishlistslice';
import deleteItemSlice from './features/slices/deleteItem';
import availablitySlice from './features/slices/changeAvailability';
import ClearCartReducer from './features/slices/clearCart';
import cartReducer from './features/slices/cart';
import addToCartReducer from './features/slices/addToCartReducer';
import removeToCartReducer from './features/slices/removeToCart';
import updateProfile from './features/slices/updateProfile';
import { orderSlice } from './features/slices/orders';
import logsSlice from './features/slices/logsSlice';
import addReviewSlice from './features/slices/addReview';
import deleteReviewSlice from './features/slices/deleteReview';
import editReviewSlice from './features/slices/editReview';
import subscribersSlice from './features/slices/subscribers';
import chatReducer from './features/slices/getMessage';
import addMessageReducer from './features/slices/addMessage';

// eslint-disable-next-line import/prefer-default-export
export const store = configureStore({
  reducer: {
    user: userReducer,
    boolean: booleanReducer,
    login: loginReducer,
    updateProfile: updateProfile.reducer,
    signup: signupReducer,
    logout: logoutReducer,
    password: passwordReducer,
    products: productSlice.reducer,
    notifications: notificationSlice,
    categories: categoriesReducer,
    addedItem: addItemReducer,
    collection: collectionSlice.reducer,
    oneproduct: oneProductSlice.reducer,
    reviews: reviewSlice.reducer,
    search: searchReducer,
    deleteItem: deleteItemSlice.reducer,
    availablity: availablitySlice.reducer,
    cart: cartReducer,
    addToCart: addToCartReducer,
    clearCart: ClearCartReducer,
    removeToCart: removeToCartReducer,
    orders: orderSlice.reducer,
    wishlist: viewWishlistSlice.reducer,
    clearWishlist: clearWishlistSlice.reducer,
    subscribers: subscribersSlice.reducer,
    addWishlist: addToWishlistSlice,
    deleteTowishlist: DeleteToWishlistSlice,
    logs: logsSlice,
    deleteReview: deleteReviewSlice,
    editReview: editReviewSlice.reducer,
    addReview: addReviewSlice,
    chat: chatReducer,
    addMessage: addMessageReducer,
  },
});
