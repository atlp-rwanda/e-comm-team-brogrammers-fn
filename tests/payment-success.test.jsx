import React from 'react';
import '@testing-library/jest-dom';
import { render, waitFor } from '@testing-library/react';
import { expect, it, describe } from '@jest/globals';
import { Provider } from 'react-redux';
import MockAdapter from 'axios-mock-adapter';
import { BrowserRouter } from 'react-router-dom';
import { store } from '../src/redux/store';
import PaymentSuccessPage, {
  ConfirmationBox,
} from '../src/Views/payments/Success';
import axios from '../src/redux/configs/axios';

const mock = new MockAdapter(axios);
const order = {
  id: '200011',
  deliveryCountry: 'Cameroon',
  deliveryCity: 'Randyberg',
  deliveryStreet: '930 Christian Dam',
  paymentMethod: 'maestro',
  orderNo: 12345,
  isPaid: false,
  status: 'Pending',
  statusUpdated: false,
  expectedDeliveryDate: '2023-05-02T13:11:37.045Z',
  totalAmount: '48100.69',
  buyerId: '2134521',
  createdAt: '2023-05-02T13:11:37.045Z',
  updatedAt: '2023-05-02T13:11:37.045Z',
  products: [
    {
      id: '10010',
      images: [
        'https://loremflickr.com/640/480',
        'https://loremflickr.com/640/480',
        'https://loremflickr.com/640/480',
      ],
      name: 'Oriental Soft Chips',
      available: true,
      price: 822,
      orderitem: {
        id: '20011',
        quantity: 84752,
        productId: '10010',
        orderId: '200011',
        statusUpdated: false,
        expectedDeliveryDate: null,
        price: 32731.16,
      },
    },
    {
      id: '10011',
      images: [
        'https://loremflickr.com/640/480',
        'https://loremflickr.com/640/480',
        'https://loremflickr.com/640/480',
      ],
      name: 'Practical Wooden Computer',
      available: true,
      price: 632,
      orderitem: {
        id: '10212',
        quantity: 77502,
        productId: '10011',
        orderId: '200011',
        statusUpdated: false,
        expectedDeliveryDate: null,
        price: 20609.87,
      },
    },
  ],
};

describe('PaymentSuccessPage', () => {
  it('should render the payment success page', async () => {
    mock.onAny().reply(200, order);
    const { getByText, getByTestId } = render(
      <Provider store={store}>
        <BrowserRouter>
          <PaymentSuccessPage />
        </BrowserRouter>
      </Provider>
    );
    await waitFor(() => {
      expect(getByText('Payment Successful')).toBeInTheDocument();
      expect(getByTestId('payment-success-icon')).toBeInTheDocument();
    });
    mock.reset();
  });
});

describe('ConfirmationBox', () => {
  const props = {
    orderNumber: order.orderNo,
    totalCost: order.totalAmount,
    deliveryDate: order.expectedDeliveryDate,
    paymentConfirmation: 'sd-2345678i9o',
  };

  it('testinng success page', () => {
    mock.onAny().reply(200, order);
    render(
      <Provider store={store}>
        <BrowserRouter>
          <ConfirmationBox {...props} />
        </BrowserRouter>
      </Provider>
    );
  });
});
