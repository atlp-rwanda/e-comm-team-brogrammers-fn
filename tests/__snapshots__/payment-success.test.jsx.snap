// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationBox should render the component 1`] = `ShallowWrapper {}`;
